core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -889,16 +870,40 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with TimeLi
    * @param stageId - The current stageId
    * @param attemptIdx - The current attempt count
    * @param numShufflePartitions - The number of partitions in the next stage
+   * @param taskSetOpt - The index of task set
+   * @param hostNames - Host on which each task in the task set is executed
+   * @param mapTaskIds - ID of each task in the task set
+   * @param execIds - ID of the executor in which each task in the task set is executed
    */
   private def completeShuffleMapStageSuccessfully(
       stageId: Int,
       attemptIdx: Int,
-      numShufflePartitions: Int): Unit = {
-    val stageAttempt = taskSets.last
+      numShufflePartitions: Int,
+      taskSetOpt: Option[Int] = None,
```

Comment:
Is this TaskSet index really needed? We should avoid rely on the ordering of TaskSets to verify the results.

Reply:
Good idea. I will improve it.